src/persist/src/operators/stream.rs

```diff
@@ -1103,59 +1061,56 @@ mod tests {
         let actual_seals: Vec<_> = listen_rx.try_iter().collect();
 
         // Assert that:
-        //  a) We don't seal primary when condition has not sufficiently advanced.
-        //  b) Condition is sealed before primary for the same timestamp OR
-        //     primary and condition got swapped when reading from two channels and a
-        //     primary seal at t is just before the condition seal at a time t' >= t.
-        //  c) We seal up, even when never receiving any data.
-        //  d) Seals happen in timestamp order.
+        //  a) We seal primary and condition to t before sealing either to t' > t. In practice,
+        //  this means we expect to see pairs of (Primary,Condition) or (Condition,Primary) in the
+        //  output, when chunked into 2-chunks.
+        //  b) We seal up, even when never receiving any data.
+        //  c) Seals happen in timestamp order.
         //
         // We cannot assert a specific seal ordering because the order is not deterministic.
 
-        let mut current_condition_seal = 0;
-        let mut current_primary_seal = 0;
         let mut condition_seals = vec![];
         let mut primary_seals = vec![];
-        let mut seal_pairs = vec![];
 
-        // Chunk up the list of seals into overlapping pairs so we can easily find
-        // the next seal.
-        for i in 0..actual_seals.len() - 1 {
-            seal_pairs.push((actual_seals[i], Some(actual_seals[i + 1])));
-        }
+        let seal_pairs = actual_seals.chunks(2);
 
-        seal_pairs.push((actual_seals[actual_seals.len() - 1], None));
+        for seal_pair in seal_pairs {
```

Comment:
I _think_ this has the potential to be flaky unfortunately. the `select!` is nondeterministic and you could imagine the  copying task gets scheduled out and then the when its scheduled in theres two seals enqueued in both the condition and primary queues. its free to e.g. read from condition twice in a row

Reply:
I think the earlier logic was more correct. is there a reason you changed it for this pr?

Reply:
Funny enough, I left the checking logic unchanged (modulo accounting for the fact that we don't seal "condition" to 4 anymore)but that made the test flaky. The version in this PR is very solid (checked with `cargo stress...`)

I'll try and see what I can do.

Reply:
The reason why the old checking logic doesn't work is because it expects that we seal _condition_ before _primary_, which we don't do anymore (we seal them atomically). An (interleaved) output like this is ok: `Primary(1), Condition(1), Primary(2), Condition(2), ...`.

I'm starting to wonder whether this test makes sense anymore, I was always wondering why it wasn't more flaky with reading from those two separate channels.